apps/aecore/src/aec_metrics_lib.erl

```diff
@@ -0,0 +1,199 @@
+parse_part(status, <<"status=", St/binary>>, Acc) ->
+    NewStatus = case St of
+                    <<"enabled">>  -> enabled;
+                    <<"disabled">> -> disabled;
+                    <<"*">>        -> '_'
+                end,
+    Acc#{status => NewStatus};
+parse_part(dps, DPsBin, #{dps := DPs} = Acc) ->
+    NewDPs = merge([binary_to_existing_atom(D,latin1)
+                    || D <- re:split(DPsBin, ",")], DPs),
+    Acc#{dps => NewDPs};
+parse_part(name, S, Acc) ->
+    Acc#{name => S}.
+
+merge([_|_] = DPs, default) ->
+    DPs;
+merge([H|T], DPs) ->
+    case lists:member(H, DPs) of
+        true  -> merge(T, DPs);
+        false -> merge(T, DPs ++ [H])
+    end;
+merge([], DPs) ->
+    DPs.
+
+find_entries_1(#{name := S, type := Type, status := Status}) ->
+    Patterns = lists:flatten([parse_stat_entry(S, Type, Status)]),
+    exometer:select(Patterns).
+
+prefix() ->
+    [ae,epoch].
+
+parse_stat_entry(E, Type, Status) when E==<<>>; E==[];
+                                       E==<<"*">>; E==<<"**">> ->
+    {{prefix() ++ '_', Type, '_'}, [{'=:=','$status',Status}], ['$_']};
+parse_stat_entry("[" ++ _ = Expr, _Type, _Status) ->
+    case erl_scan:string(ensure_trailing_dot(Expr)) of
+        {ok, Toks, _} ->
+            case erl_parse:parse_exprs(Toks) of
+                {ok, [Abst]} ->
+                    partial_eval(Abst);
+                Error ->
+                    io:fwrite("(Parse error for ~p: ~p~n", [Expr, Error]),
+                    []
+            end;
+        ScanErr ->
+            io:fwrite("(Scan error for ~p: ~p~n", [Expr, ScanErr]),
+            []
+    end;
+parse_stat_entry(Str, Type, Status) when Status==enabled; Status==disabled ->
+    Parts = re:split(Str, "\\.", [{return,list}]),
+    Heads = replace_parts(Parts),
+    [{{H,Type,Status}, [], ['$_']} || H <- Heads];
+parse_stat_entry(Str, Type, '_') ->
+    Parts = re:split(Str, "\\.", [{return,list}]),
+    Heads = replace_parts(Parts),
+    [{{H,Type,'_'}, [], ['$_']} || H <- Heads];
+parse_stat_entry(_, _, Status) ->
+    io:fwrite("(Illegal status: ~p~n", [Status]).
+
+ensure_trailing_dot(Str) ->
+    case lists:last(Str) of
+        $. ->
+            Str;
+        _ ->
+            Str ++ "."
+    end.
+
+partial_eval({cons,_,H,T}) ->
+    [partial_eval(H) | partial_eval(T)];
+%% partial_eval({nil,_}) ->
+%%     [];
+partial_eval({tuple,_,Elems}) ->
+    list_to_tuple([partial_eval(E) || E <- Elems]);
+%% partial_eval({T,_,X}) when T==atom; T==integer; T==float ->
+%%     X;
+partial_eval({op,_,'++',L1,L2}) ->
+    partial_eval(L1) ++ partial_eval(L2);
+partial_eval(X) ->
+    erl_parse:normalise(X).
+
+replace_parts(Parts) ->
+    case split("**", Parts) of
+        {_, []} ->
+            [replace_parts_1(Parts)];
+        {Before, After} ->
+            Head = replace_parts_1(Before),
+            Tail = replace_parts_1(After),
+            [Head ++ Pad ++ Tail || Pad <- pads()]
+    end.
+
+pads() ->
```

Comment:
Really? Would a function pad(N) -> [ '_'  || _ lists:seq(1,N)] not be nicer and then pads(K) -> [ pads(N) || N <- lists:seq(1,K)].

Reply:
Yes ... that was copied from riak_core, and those guys were pretty focused on speed.
My favorite solution would be to extent ets:select() match specs. :)